server/cluster/coordinator.go

```diff
@@ -966,3 +967,17 @@ func (c *coordinator) getPausedSchedulerDelayUntil(name string) (int64, error) {
 	}
 	return s.GetDelayUntil(), nil
 }
+
+func (c *coordinator) CheckTransferWitnessLeader(region *core.RegionInfo) {
+	c.RLock()
+	defer c.RUnlock()
+	if s, ok := c.schedulers[schedulers.TransferWitnessLeaderName]; ok {
```

Comment:
I suggest releasing the read lock as soon as possible here since the `select` below may have the possibility to block.

```suggestion
	c.RLock()
	scheduler, ok := c.schedulers[schedulers.TransferWitnessLeaderName];
	c.RUnlock()
```

Reply:
Fixed, thanks!